optimize/backend/src/it/java/io/camunda/optimize/test/it/extension/db/OpenSearchDatabaseTestService.java

```diff
@@ -903,6 +767,149 @@ public VariableUpdateInstanceIndex getVariableUpdateInstanceIndex() {
+        getMappingResponse.result().values().stream()
+            .findFirst()
+            .orElseThrow(
+                () ->
+                    new OptimizeRuntimeException(
+                        "There should be at least one mapping available for the index!"))
+            .mappings()
+            .properties();
+    if (propertiesMap instanceof Map) {
+      return (Map<String, Object>) propertiesMap;
+    } else {
+      throw new OptimizeRuntimeException("Database index mapping properties should be of type map");
+    }
+  }
+
+  @Override
+  public boolean indexExists(final String indexOrAliasName, Boolean addMappingFeatures) {
+    return indexExists(indexOrAliasName);
+  }
+
+  @Override
+  public boolean templateExists(final String optimizeIndexTemplateNameWithVersion)
+      throws IOException {
+    final ExistsTemplateRequest.Builder request =
+        new ExistsTemplateRequest.Builder().name(optimizeIndexTemplateNameWithVersion);
+    return getOptimizeOpenSearchClient()
+        .getOpenSearchClient()
+        .indices()
+        .existsTemplate(request.build())
+        .value();
+  }
+
+  @Override
+  public boolean isAliasReadOnly(final String readOnlyAliasForIndex) throws IOException {
+    final GetAliasResponse aliases =
+        getOptimizeOpenSearchClient()
+            .getAlias(
+                GetAliasRequest.of(
+                    a ->
+                        a.name(
+                            getOptimizeOpenSearchClient()
+                                .applyIndexPrefixes(readOnlyAliasForIndex))));
+    return aliases.result().values().stream()
+        .flatMap(a -> a.aliases().values().stream())
+        .collect(Collectors.toSet())
+        .stream()
+        .noneMatch(AliasDefinition::isWriteIndex);
+  }
+
+  @Override
+  public List<String> getAllIndicesWithReadOnlyAlias(final String aliasNameWithPrefix) {
+    final GetAliasResponse aliasResponse;
+    try {
+      aliasResponse = getOptimizeOpenSearchClient().getAlias(aliasNameWithPrefix);
+    } catch (final IOException e) {
+      throw new OptimizeRuntimeException(e);
+    }
+    final Map<String, IndexAliases> indexNameToAliasMap = aliasResponse.result();
+    return indexNameToAliasMap.entrySet().stream()
+        .filter(
+            entry ->
+                entry.getValue().aliases().values().stream()
+                    .anyMatch(alias -> alias.isWriteIndex() != null && !alias.isWriteIndex()))
+        .map(Map.Entry::getKey)
+        .toList();
+  }
+
+  private IndexSettings createIndexSettings(
+      final IndexMappingCreator indexMappingCreator,
+      final ConfigurationService configurationService) {
+    try {
+      return OpenSearchIndexSettingsBuilder.buildAllSettings(
+          configurationService, indexMappingCreator);
+    } catch (final IOException e) {
+      throw new OptimizeRuntimeException("Could not create index settings");
+    }
+  }
+
+  public boolean indexExists(final String indexOrAliasName) {
```

Comment:
this is the actual change, the rest was just re-sorting done by the save-actions file